GooglePlayInstant/Editor/PlayInstantQuickDeployInputManager.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using UnityEditor;
+
+namespace GooglePlayInstant.Editor
+{
+    /// <summary>
+    /// Provides functionality for reading, storing and retrieving text field values from "Deploy Bundle" section.
+    /// </summary>
+    public static class DeployBundleTextFields
+    {
+        private const string DataStoragePrefix = "GooglePlayInstant.Editor.QuickDeploy.";
+        private static string _oauth2CredentialsPath;
+        private static string _localAssetBundlePath;
+        private static string _remoteBucketName;
+        private static string _remoteObjectName;
```

Comment:
Rather than have these 4 fields, consider using a Dictionary where the key is the same as you use for the preference file (e.g. "oauth2CredentialsPath") and the value is the cached value. This would allow you to share a lot more of the get/set code below (e.g. null and equality checks would be shared).

Note: please see my comment below about performance optimization. This local caching should only be done if you've proven that it's necessary.

Reply:
Resolved. Abandoned this code since the config pull request addresses this